kore/src/Kore/Internal/Inj.hs

```diff
@@ -0,0 +1,143 @@
+{-|
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+
+module Kore.Internal.Inj
+    ( Inj (..)
+    , toSymbol, toApplication
+    -- * Exceptions
+    , UnorderedInj (..)
+    , unorderedInj
+    ) where
+
+import Control.DeepSeq
+import Control.Exception
+    ( Exception (..)
+    , throw
+    )
+import Data.Function
+    ( (&)
+    )
+import qualified Data.Function as Function
+import Data.Hashable
+import qualified Data.Text.Prettyprint.Doc as Pretty
+import Data.Typeable
+    ( Typeable
+    )
+import qualified Generics.SOP as SOP
+import qualified GHC.Generics as GHC
+
+import Debug
+import Kore.Attribute.Pattern.FreeVariables
+    ( FreeVariables
+    )
+import qualified Kore.Attribute.Symbol as Attribute
+import Kore.Attribute.Synthetic
+    ( Synthetic (..)
+    )
+import Kore.Internal.ApplicationSorts
+import Kore.Internal.Symbol
+import Kore.Sort
+import Kore.Syntax.Application
+    ( Application (..)
+    )
+import Kore.Unparser
+
+data Inj a =
+    Inj
+        { injConstructor :: !Id
+        , injFrom        :: !Sort
+        , injTo          :: !Sort
+        , injAttributes  :: !Attribute.Symbol
+        , injChild       :: !a
+        }
+    deriving (GHC.Generic, Show)
+    deriving (Functor, Foldable, Traversable)
+
+instance Eq a => Eq (Inj a) where
+    (==) a b =
```

Comment:
Here and below, why not match on `(Inj _ _ _ _ _)` in the same way as you do for `toSymbol`?